pkg/controller/pxc/controller.go

```diff
@@ -601,21 +601,24 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 			return err
 		}
 
-		// TODO: code duplication with updatePod function
-		if proxySet.Spec.Template.Annotations == nil {
-			proxySet.Spec.Template.Annotations = make(map[string]string)
-		}
+
+		proxySet.Spec.Template.Annotations = make(map[string]string)
```

Comment:
what if there are already some data in proxySet.Spec.Template.Annotations?

Reply:
They will be in `cr` =)
Our service annotations will be filled below

Reply:
but for example we use annotations here https://github.com/percona/percona-xtradb-cluster-operator/blob/master/pkg/controller/pxc/users.go#L227